internal/core/chats/chat.repository.go

```diff
@@ -0,0 +1,88 @@
+package chats
+
+import (
+	"database/sql"
+	"time"
+
+	"github.com/brain-flowing-company/pprp-backend/internal/models"
+	"github.com/google/uuid"
+	"gorm.io/gorm"
+)
+
+type Repository interface {
+	GetAllChats(*[]models.ChatPreviews, uuid.UUID) error
+	GetMessagesInChat(*[]models.Messages, uuid.UUID, uuid.UUID, int, int) error
+	SaveMessages(msg *models.Messages) error
+	ReadMessages(sendUserId uuid.UUID, recvUserId uuid.UUID) error
+}
+
+type repositoryImpl struct {
+	db *gorm.DB
+}
+
+func NewRepository(db *gorm.DB) Repository {
+	return &repositoryImpl{
+		db,
+	}
+}
+
+func (repo *repositoryImpl) GetAllChats(results *[]models.ChatPreviews, userId uuid.UUID) error {
+	_ = repo.db.Model(&models.Messages{}).
```

Comment:
I think the query is syntactically correct, but it might result in an error somehow. Would it be preferable to return the error message instead?